### ISSUE TITLE
[FEATURE] Add `databricks_notification_destination` data source

### ISSUE THREAD

**Issue by mgyucht** [databricks/terraform-provider-databricks#3950]
### Use-cases
For users who want to make use of notification destinations that are created out-of-band or in other terraform templates, it would be convenient to add a data source for notification destinations. This would be useful in DABs for example, where notification destinations may be configured once by a workspace admin, and then used by many users in their various jobs or pipelines.

### Proposal
```hcl
data databricks_notification_destinations {} # Lists all notification destinations

data databricks_notification_destinations {
  display_name = "..."
  type = "..."
}
```

**Comment by VOVELEE** [comment]
Hey, 

 the APIs are finally released: https://docs.databricks.com/api/azure/workspace/notificationdestinations/create
I am also interested in this :)

**Comment by dgomez04** [comment]
@mgyucht closed by #4087

**Comment by alexott** [comment]
Fixed in https://github.com/databricks/terraform-provider-databricks/pull/4087
